### PR TITLE
Add New API to config merge

### PR DESCRIPTION
**Pull request by sarao1310**
- [ ] All [tests](https://github.com/jfrog/jfrog-client-go#tests) passed. If this feature is not already covered by the tests, I added new tests.
- [ ] All [static analysis checks](https://github.com/jfrog/jfrog-client-go/actions/workflows/analysis.yml) passed.
- [x] This pull request is on the dev branch.
- [ ] I used gofmt for formatting the code before submitting the pull request.
-----

### CODE REVIEW

**Review by eyalbe4 — CHANGES_REQUESTED** [review]
* The tests ate falling. Please ensure to merge the PR only when all is green,
* The README should include the new APIs added.
* Since this PR wasn't properly rebased, it includes the code of other PRs as well, which made it harder to review. See how you can avoid that in the future.
* The PR name is incorrect, and does not reflect its content at all. The name you chose for the PR describes the goal for your woek, which not relevant for the scope of this PR.
* You should add a proper label for the PR.

**Comment by yahavi** [comment]
@sarao1310 the base branch is "master". Please change it to dev.

File: artifactory/services/utils/artifactoryutils.go [hunk 1]
```diff
@@ -139,6 +139,16 @@ func IsSubPath(paths []string, index int, separator string) bool {
 	return false
 }
 
+// Returns true if one or more paths have a 'prefix' as a prefix
```
[the remark this answers is not in the document]
Comment (sarao1310) [reply artifactory/services/utils/artifactoryutils.go]: Done

**Review by eyalbe4 — CHANGES_REQUESTED** [review]
1. The PR name is incorrect, and does not reflect its content at all. The name you chose for the PR describes the goal for your woek, which not relevant for the scope of this PR. I suggest the following name -  **New 'CreateRepositoryWithParams' and 'GetAllProjects' APIs**

2. The README should include the new APIs added.

3. You should add a proper label for the PR.

**Pull request closed without merging**